src/util/address.rs

```diff
@@ -672,6 +672,13 @@ impl Address {
 
     /// Constructs an [`Address`] from an output script (`scriptPubkey`).
     pub fn from_script(script: &script::Script, network: Network) -> Option<Address> {
+        if script.is_witness_program() {
+            // Segwit v0 script length check.
```

Comment:
Can we leave this comment out please, it says what the code is doing but we can see this in the next line of code. This suggestions is based on the idea that comments should say _why_ and code says _what_.

Although, since the 22 and 34 are magic numbers that assume Bitcoin knowledge, perhaps this would be better
```rust
    /// Constructs an [`Address`] from an output script (`scriptPubkey`).
    pub fn from_script(script: &Script, network: Network) -> Option<Address> {
        if script.is_witness_program() && !is_valid_segwit_v0_script_length(&script) {
            return None
        }

        Some(Address {
            payload: Payload::from_script(script)?,
            network,
        })
    }

...

fn is_valid_segwit_v0_script_length(script: &Script) -> bool {
    script.len() == 22 || script.len() == 34
}
```

Reply:
I agree that the comment is fairly pointless but I didn't think creating a new function just to address the magic numbers was the best way to go so I have instead opted to use the `script.is_v0_p2wpkh()` and  `script.is_v0_p2wsh()` functions as they check the length of the script as well as if they are valid v0 scripts in general.

Reply:
Nice, I like it!